cmd/juju/charmhub/data.go

```diff
@@ -166,6 +167,34 @@ func channelArches(platforms []charmhub.Platform) set.Strings {
 	return arches
 }
 
+func filterFindResults(in []charmhub.FindResponse, architecture, series string) []charmhub.FindResponse {
+	allArch := architecture == ArchAll
+	allSeries := series == SeriesAll
+
+	// If we're searching for everything then we can skip the filtering logic
+	// and return immediately.
+	if allArch && allSeries {
+		return in
+	}
+
+	witnessed := make([]charmhub.FindResponse, 0)
+
+	for _, resp := range in {
```

Comment:
I'd love to standardise this, is it worth making a new type that does this. Seems a bit wasteful?